client/src/context/AuthContext.js

```diff
@@ -0,0 +1,86 @@
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
+import createDataContext from "./createDataContext";
+import trackerApi from "../api/tracker";
+
+const authReducer = (state, action) => {
+  switch (action.type) {
+    case "add_error":
```

Comment:
For these strings. Since you are using JavaScript you are at disadvantage. With a small typo you can have an error and you are lucky if you find it quickly.

Usually with any typed language you would put these texts inside an enum. But you don't have an enum in JS.

What you could do however, is put these inside an object. And reference the values by accessing them from inside of an object.